src/modules/fw_pos_control_l1/FixedwingPositionControl.hpp

```diff
@@ -362,74 +422,193 @@ class FixedwingPositionControl final : public ModuleBase<FixedwingPositionContro
 	 * @brief Moves the current position setpoint to a value far ahead of the current vehicle yaw when in  a VTOL
 	 * transition.
 	 *
-	 * @param[in,out] current_sp current position setpoint
+	 * @param[in,out] current_sp Current position setpoint
 	 */
 	void move_position_setpoint_for_vtol_transition(position_setpoint_s &current_sp);
 
-	uint8_t		handle_setpoint_type(const uint8_t setpoint_type, const position_setpoint_s &pos_sp_curr);
-	void		control_auto(const float control_interval, const Vector2d &curr_pos, const Vector2f &ground_speed,
-				     const position_setpoint_s &pos_sp_prev, const position_setpoint_s &pos_sp_curr, const position_setpoint_s &pos_sp_next);
+	/**
+	 * @brief Changes the position setpoint type to achieve the desired behavior in some instances.
+	 *
+	 * @param pos_sp_curr Current position setpoint
+	 * @return Adjusted position setpoint type
+	 */
+	uint8_t	handle_setpoint_type(const position_setpoint_s &pos_sp_curr);
+
+	/* automatic control methods */
+
+	/**
+	 * @brief Position control for all automatic modes except takeoff and landing
+	 *
+	 * @param control_interval Time since last position control call [s]
+	 * @param curr_pos Current 2D local position vector of vehicle [m]
+	 * @param ground_speed Local 2D ground speed of vehicle [m/s]
+	 * @param pos_sp_prev previous position setpoint
+	 * @param pos_sp_curr current position setpoint
+	 * @param pos_sp_next next position setpoint
+	 */
+	void control_auto(const float control_interval, const Vector2d &curr_pos, const Vector2f &ground_speed,
+			  const position_setpoint_s &pos_sp_prev, const position_setpoint_s &pos_sp_curr, const position_setpoint_s &pos_sp_next);
+
+	/**
+	 * @brief Controls altitude and airspeed for a fixed-bank loiter.
```

Comment:
Could add that this is meant as a failsafe mode after lateral position estimation failure.